WeatherApp-UIKit/WeatherApp-UIKitTests/TestURLProtocol.swift

```diff
@@ -0,0 +1,79 @@
+//
+//  TestURLProtocol.swift
+//  WeatherApp-UIKitTests
+//
+//  Created by Vojta on 04/01/2020.
+//  Copyright © 2020 Industrial Binaries. All rights reserved.
+//
+
+import Foundation
+
+
+/// The URLProtocol subclass allowing to intercept the network communication
+/// and provide custom mock responses for the given URLs.
+class TestURLProtocol: URLProtocol {
+
+  /// The key-pairs of URLs this URLProtocol intercepts with their simulated response.
+  static var mockResponses: [URL: (URLRequest) -> (result: Result<Data, Error>, statusCode: Int?)] = [:]
+
+  override class func canInit(with request: URLRequest) -> Bool {
+    guard let url = request.url else { return false }
+    return mockResponses.keys.contains(url.withoutQueryComponents)
+  }
+
+  override class func canonicalRequest(for request: URLRequest) -> URLRequest {
+    // Overriding this function is required by the superclass.
+    return request
+  }
```

Comment:
Maybe move it on bottom of class to `stopLoading` when this method is not important of us

Reply:
👍